sql/lock.cc

```diff
@@ -1096,15 +1096,18 @@ void Global_read_lock::unlock_global_read_lock(THD *thd) {
 
     /* If node not part of the cluster avoid running the resume/pause action */
     if (server_state.state() != Wsrep_server_state::s_disconnected) {
+      /*
+       Resume wsrep server even if current thread has disabled replication.
+       We allow other nodes to replicate data to this server.
+       */
       if (server_state.state() == Wsrep_server_state::s_donor ||
-          (wsrep_on(thd) &&
-           server_state.state() != Wsrep_server_state::s_synced)) {
+          server_state.state() != Wsrep_server_state::s_synced) {
```

Comment:
Isn't this if now basically `if(server_state.state() != Wsrep_server_state::s_synced)`? Do we really want to do this even in disconnected/initializing state?

And what if we built the server with WITH_WSREP, but aren't running a cluster? Do normal, non galera ftwrl tests pass after this change?

Reply:
This part of code is not executed when server is disconnected (see 1 "if" above)
I have checked all MTR tests which use FTWRL. No issues noticed.